dialog-db/balancing-distribution.md

```diff
@@ -0,0 +1,785 @@
+    [*] --> g_:::index
+    [*] --> h_:::index
+    [*] --> p_:::index
+
+    a_ --> a
+    a_ --> b
+    a_ --> c:::new
+    a_ --> d
+    a_ --> e
+    
+    g_ --> g:::weak
+    
+    h_ --> h:::weak
+    h_ --> i
+    h_ --> j
+    h_ --> k
+    h_ --> m
+    h_ --> n
+    
+    p_ --> p:::strong
+    p_ --> q
+    p_ --> r
+    p_ --> t
+    
+```
+
+Now if we insert `f` we'll end up with pressure to split again and with another sibling group
+
+```mermaid
+---
+config:
+  theme: 'base'
+  themeVariables:
+    primaryColor: '#9398B0'
+    primaryTextColor: '#fff'
+    primaryBorderColor: '#9398B0'
+    secondaryBorderColor: '#9398B0'
+    lineColor: '#9398B0'
+    secondaryColor: '#000000'
+    secondaryTextColor: '#000000'
+    tertiaryColor: '#fff'
+---
+stateDiagram
+		classDef weak fill:#FFC034,stroke-width:0;
+		classDef strong fill:#E599F7,stroke-width:0;
+		classDef index fill:#4DABF7,stroke-width:0;
+		classDef new fill:#099268,stroke-width:0;
+		classDef delete fill:#4DABF7,stroke-width:0,opacity: 0.3, border:0;
+
+    a_: a..e
+    f_: f..
+    g_: g..
+    h_: h..n
+    p_: p..t
+    
+    [*] --> a_:::index
+    [*] --> f_:::index
+    [*] --> g_:::index
+    [*] --> h_:::index
+    [*] --> p_:::index
+
+    a_ --> a
+    a_ --> b
+    a_ --> c:::new
+    a_ --> d
+    a_ --> e
+    
+    f_ --> f:::weak
+    g_ --> g:::weak
+    
+    h_ --> h:::weak
+    h_ --> i
+    h_ --> j
+    h_ --> k
+    h_ --> m
+    h_ --> n
+    
+    p_ --> p:::strong
+    p_ --> q
+    p_ --> r
+    p_ --> t
+    
+```
+
+If have switched insertion order and did `f` then `c` we would have with a different tree instead 
+
+```mermaid
+---
+config:
+  theme: 'base'
+  themeVariables:
+    primaryColor: '#9398B0'
+    primaryTextColor: '#fff'
+    primaryBorderColor: '#9398B0'
+    secondaryBorderColor: '#9398B0'
+    lineColor: '#9398B0'
+    secondaryColor: '#000000'
+    secondaryTextColor: '#000000'
+    tertiaryColor: '#fff'
+---
+stateDiagram
+		classDef weak fill:#FFC034,stroke-width:0;
+		classDef strong fill:#E599F7,stroke-width:0;
+		classDef index fill:#4DABF7,stroke-width:0;
+		classDef new fill:#099268,stroke-width:0;
+		classDef delete fill:#4DABF7,stroke-width:0,opacity: 0.3, border:0;
+
+    a_: a..e
+    f_: f..g
+    h_: h..n
+    p_: p..t
+    
+    [*] --> a_:::index
+    [*] --> f_:::index
+    [*] --> h_:::index
+    [*] --> p_:::index
+
+    a_ --> a
+    a_ --> b
+    a_ --> c:::new
+    a_ --> d
+    a_ --> e
+    
+    f_ --> f:::weak
+    f_ --> g
+    
+    h_ --> h:::weak
+    h_ --> i
+    h_ --> j
+    h_ --> k
+    h_ --> m
+    h_ --> n
+    
+    p_ --> p:::strong
+    p_ --> q
+    p_ --> r
+    p_ --> t
+    
+```
+
+
+
+
+
+> I do not know which one dolt compromises here, but it must be one or the other, because insertion may split a sibling group at which point split out segment either joins the next group causing potentially cascading [boundary shifts][] or stay independent compromising [history independence][].
```

Comment:
I really enjoy these "asides" throughout, makes it feel less academic and more personal.